ports/esp32/main.c

```diff
@@ -190,6 +190,8 @@ void mp_task(void *pvParameter) {
     mp_bluetooth_deinit();
     #endif
 
+    network_wlan_deinit_wifi();
```

Comment:
This is a big, breaking change to user workflow.  The idea is to keep the WiFi up and connected to an AP so that it doesn't have to be reconnected after a soft reset.  This is very useful for running a set of network tests, for example.

So I suggest not to make this change.

Reply:
Ok - I understand and appreciate that perspective.

I'll revert it, though personally I'd put more weight on reproducibilty :-). My experience is that many users are confused on this point, which leads to perceptions of unreliability or lack of reproducibility - especially with more complex wifi configurations. For instance, it has been a source of preceived issues among users of the espnow PR.

Instead I will look into a separate PR for some more expansive wifi docs for the ESP32 and ESP8266 (users are also confused on other points wrt wifi config on these devices).

Reply:
@dpgeorge Do you have interest in the other parts of this PR - ie. to automatically shutdown the wifi on light/deep sleep?

If so, I'll update accordingly. It is a viable option to leave that as a user's responsibility, but I have found that many users are unaware of this requirement. In practice, you can often get away with not stopping the wifi before sleep, but it causes problems often enough that users perceive it as unreliability or a bug.

Reply:
> ie. to automatically shutdown the wifi on light/deep sleep?

I'm not sure... other ports (eg stm32 with cyw43, and Pico W) don't need to shutdown wifi when doing a sleep, so it would be inconsistent if esp32 did it.  Also it seems that turning off wifi is a blocking operation that can take a while, so a short (eg <100ms) lightsleep could block for a lot longer than expected.

I'm not sure exactly how esp32 low-power modes work, but maybe there's a different hardware lightsleep that can be used which does retain wifi?  Alternatively we could make lightsleep just do a normal time.sleep if wifi is enabled?

Reply:
OK - I'll do some experiments to determine the impacts, so please consider this PR on hold for now. The advice from Espressif is that "Before entering Deep-sleep or Light-sleep modes, the application must disable Wi-Fi and Bluetooth using the appropriate calls." However, I have found that if you don't do this, after deep/light sleep the wifi is often still usable, but sometimes the peripheral is no longer usable (eg. repeatedly fails to connect) and a hard reset is necessary.